week10/pages/Detail.jsx

```diff
@@ -0,0 +1,159 @@
+import { useState, useEffect } from "react";
+import { PostDetail, EditPost, DeletePost } from "../api/post";
+import { PutHeart, DeleteHeart, CntHeart } from "../api/heart";
+import { CreateComment, DeleteComment, GetComment } from "../api/comment";
+import styled from "styled-components";
+import { useParams, useNavigate } from "react-router-dom";
+const Title = styled.div``;
+const Content = styled.div``;
+const Image = styled.img`
+  height: 300px;
+  width: 300px;
+`;
+const CommentList = styled.div``;
+const CommentItem = styled.div``;
+const Detail = () => {
+  const { postId } = useParams();
+  const [post, setPost] = useState(null);
+  const [likeCnt, setLikeCnt] = useState(0);
+  const [comment, setComment] = useState();
+  const [comments, setComments] = useState([]);
+  const [commentCnt, setCommentCnt] = useState(0);
+  const [isEditing, setIsEditing] = useState(false);
+  const [editedTitle, setEditedTitle] = useState("");
+  const [editedContent, setEditedContent] = useState("");
+  const navigate = useNavigate();
+  useEffect(() => {
+    const fetchPostDetail = async () => {
+      try {
+        const data = await PostDetail(postId);
+        setPost(data);
+        setEditedTitle(data.title);
+        setEditedContent(data.content);
+        const heartData = await CntHeart(postId);
+        setLikeCnt(heartData);
+        const commentData = await GetComment(postId);
+        setComments(commentData);
+        setCommentCnt(commentData.length);
+      } catch (err) {
+        console.log(err);
+      }
+    };
+    fetchPostDetail();
+  }, [postId]);

```

Comment:
P5: api 호출이 여러번 발생하네요, promise.all을 사용해서 호출 횟수를 줄이고, 최적화를 하면 좋을 것 같아요!

```suggestion
useEffect(() => {
  const fetchPostDetail = async () => {
    try {
      const postData = await PostDetail(postId);
      setPost(postData);
      setEditedPost({ title: postData.title, content: postData.content });

      const [heartData, commentData] = await Promise.all([
        CntHeart(postId),
        GetComment(postId),
      ]);

      setInteraction({
        likeCount: heartData,
        commentCount: commentData.length,
      });
      setComments(commentData);
    } catch (err) {
      console.error("Failed to fetch post details", err);
    }
  };

  fetchPostDetail();
}, [postId]);


```